pan3d/dataset_builder.py

```diff
@@ -1,624 +1,291 @@
-        self.server = server
-        self._layout = None
-        self._force_local_rendering = not has_gpu_rendering()
-
-        self.state.update(initial_state)
-        self.algorithm = PyVistaXarraySource()
-        self.plotter = pyvista.Plotter(off_screen=True, notebook=False)
-        self.plotter.set_background("lightgrey")
-        self.dataset = None
-        self.da = None
-        self._mesh = None
-        self.actor = None
-
-        self.ctrl.get_plotter = lambda: self.plotter
-        self.ctrl.reset = self.reset
-
-        if pangeo:
-            with open(Path(BASE_DIR, "../examples/pangeo_catalog.json")) as f:
-                self.state.available_datasets += json.load(f)
+        self._server = server
+        self._pangeo = pangeo
 
-        if dataset_path:
-            self.state.dataset_path = dataset_path
-            self.set_dataset_path(dataset_path=dataset_path)
-        if state:
-            self.state.update(state)
-
-        if self._force_local_rendering:
-            pyvista.global_theme.trame.default_mode = "client"
+        self.dataset_path = dataset_path
 
     # -----------------------------------------------------
     # Properties
     # -----------------------------------------------------
 
     @property
-    def state(self) -> trame_server.state.State:
-        """Returns the current State of the Trame server."""
-        return self.server.state
-
-    @property
-    def ctrl(self) -> trame_server.controller.Controller:
-        """Returns the Controller for the Trame server."""
-        return self.server.controller
-
-    @property
-    def data_array(self) -> xarray.core.dataarray.DataArray:
-        """Returns the current Xarray data array with current slicing applied."""
-        return self.algorithm.sliced_data_array
-
-    @property
-    def mesh(
-        self,
-    ) -> typing.Union[pyvista.core.grid.RectilinearGrid, pyvista.StructuredGrid]:
-        """Returns the PyVista Mesh derived from the current data array."""
-        if self._mesh is None:
-            self._mesh = self.algorithm.mesh
-        return self._mesh
+    def viewer(self):
+        from pan3d.dataset_viewer import DatasetViewer
+
+        if self._viewer is None:
+            self._viewer = DatasetViewer(
+                builder=self,
+                server=self._server,
+                pangeo=self._pangeo,
+                state=dict(
+                    dataset_path=self.dataset_path,
+                    da_active=self.data_array_name,
+                    da_x=self.x,
+                    da_y=self.y,
+                    da_z=self.z,
+                    da_t=self.t,
+                    da_t_index=self.t_index,
+                ),
+            )
+            self._viewer._dataset_changed()
+            self._viewer._data_array_changed()
+            self._viewer._time_index_changed()
+            self._viewer._mesh_changed()
+        return self._viewer
```

Comment:
Should all the 
```
self._viewer._dataset_changed()
self._viewer._data_array_changed()
self._viewer._time_index_changed()
self._viewer._mesh_changed()
```
be done automatically in the constructor?

Reply:
Sure. They'll all be no-ops in the typical instantiation case (not through the builder), but I agree moving them there is more consistent. Changed in [b87a945](https://github.com/Kitware/pan3d/pull/59/commits/b87a9450a989c0c672d41afe7d852b030323d219)